client/keys/add.go

```diff
@@ -117,56 +118,66 @@ func RunAddCmd(ctx client.Context, cmd *cobra.Command, args []string, inBuf *buf
 		return err
 	}
 
-	if dryRun, _ := cmd.Flags().GetBool(flags.FlagDryRun); !dryRun {
-		_, err = kb.Key(name)
-		if err == nil {
-			// account exists, ask for user confirmation
-			response, err2 := input.GetConfirmation(fmt.Sprintf("override the existing name %s", name), inBuf, cmd.ErrOrStderr())
-			if err2 != nil {
-				return err2
-			}
+	_, err = kb.Key(name)
+	if err == nil {
+		// account exists, ask for user confirmation
+		response, err2 := input.GetConfirmation(fmt.Sprintf("override the existing name %s", name), inBuf, cmd.ErrOrStderr())
+		if err2 != nil {
+			return err2
+		}
 
-			if !response {
-				return errors.New("aborted")
-			}
+		if !response {
+			return errors.New("aborted")
+		}
 
-			err2 = kb.Delete(name)
-			if err2 != nil {
-				return err2
-			}
+		if dryRun {
+			return errors.New("override not available with --dry-run")
```

Comment:
Adding this here to ensure no keys can be overwritten with `--dry-run`.

This was not necessary before because we were skipping both the override block and the multisig block. Now the confirmation message appears but responding yes will throw this error.